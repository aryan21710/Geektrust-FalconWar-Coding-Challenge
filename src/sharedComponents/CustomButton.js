import React, { useState } from 'react';
import { Button, ButtonText, AnimatedMiniJet } from './styles';
import { useHistory } from 'react-router';
import { useSpring, config } from 'react-spring';
import { MinijetImage } from '../lib/constants';
import PropTypes from 'prop-types';

export const CustomButton = (props) => {
	const { redirectPath, leftpos, TextForButton, width, disabled, opacity } = props;
	const [isHover, setIshover] = useState(false);
	const animateJet = () => setIshover(true);
	const unAnimateJet = () => setIshover(false);
	const changePageOnClick = () => !disabled && history.push(`${redirectPath}`);
	const history = useHistory();
	const { Minijet } = MinijetImage;

	const jetAnimatedProp = useSpring({
		transform: isHover && !disabled ? 'translateX(6vw)' : 'translateX(-30vw)',
		config: config.stiff,
	});

	const btnTextProp = useSpring({
		opacity: isHover && !disabled ? 0 : 1,
		config: config.stiff,
	});

	return (
		<React.Fragment>
			<Button
				disabled={disabled}
				width={width}
				onMouseEnter={animateJet}
				onMouseLeave={unAnimateJet}
				onClick={changePageOnClick}
				opacity={opacity}
				id="redirectToNextPageBtn"
			>
				<AnimatedMiniJet id="CustomButton-MiniJetImg"leftpos={leftpos} style={jetAnimatedProp} src={Minijet} />
				<ButtonText style={btnTextProp}>{TextForButton}</ButtonText>
			</Button>
		</React.Fragment>
	);
};

CustomButton.propTypes = {
	redirectPath: PropTypes.string,
	leftpos: PropTypes.string,
	TextForButton: PropTypes.string,
	width: PropTypes.string,
	disabled: PropTypes.bool,
	opacity: PropTypes.number,
};
