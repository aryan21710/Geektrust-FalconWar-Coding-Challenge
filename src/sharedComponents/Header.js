import React from 'react';
import { Heading, HeaderWrapper } from './styles';

const Header = () => {
	return (
		<HeaderWrapper>
			<Heading data-testid="Headerh1">WELCOME TO THE BATTLE OF TARA B</Heading>
		</HeaderWrapper>
	);
};

export default Header;
