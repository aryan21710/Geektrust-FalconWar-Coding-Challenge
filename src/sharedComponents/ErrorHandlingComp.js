import React from 'react';
import { Wrapper, Heading, Button, ButtonText, BadgeWrapper } from './styles';
import { CustomButton } from './CustomButton';
import { withRouter } from 'react-router-dom';

class ErrorHandlerComp extends React.Component {
	constructor(props) {
		super(props);
		this.state = { error: null, errorInfo: null };
	}

	componentDidCatch(error, errorInfo) {
		this.setState({
			error: error,
			errorInfo: errorInfo,
		});
	}

	render() {
		if (this.state.errorInfo) {
			return (
				<Wrapper justifyContent="space-around" flexDirection="column">
					<Heading>OOPS Something went wrong.</Heading>
					<br />
					<br />
					<BadgeWrapper flex="0" flexDirection="column"  width="100vw" style={{ whiteSpace: 'pre-wrap' }}>
						<Heading fontFamily="Arial" fontSize="1.2rem" >
							{this.state.error && this.state.error.toString()}
						</Heading>
						<br />
						<Heading fontFamily="Arial" fontSize="1.2rem">
							{this.state.errorInfo.componentStack}
						</Heading>
					</BadgeWrapper>
					<Button
						onClick={() => {
							this.props.history.push('/');
							window.location.reload();
						}}
					>
						<ButtonText>BACK TO HOME</ButtonText>
					</Button>
				</Wrapper>
			);
		}
		return this.props.children;
	}
}

export default withRouter(ErrorHandlerComp);
