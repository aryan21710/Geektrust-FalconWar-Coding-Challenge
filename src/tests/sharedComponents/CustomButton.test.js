import React from 'react';
import { CustomButton } from '../../sharedComponents/CustomButton';
import { shallow, mount, render } from '../../../enzyme';
import toJSON from 'enzyme-to-json';


describe('TESTING CustomButton COMPONENT', () => {
	test('Snapshot', () => {
		const [redirectPath, leftpos, TextForButton, width, disabled, opacity] = [
			'someText',
			'someText',
			'someText',
			'someText',
			true,
			1,
        ];
        
		const wrapper = shallow(
			<CustomButton
				redirectPath={redirectPath}
				leftpos={leftpos}
				TextForButton={TextForButton}
				width={width}
				disabled={disabled}
				opacity={opacity}
			/>
		);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	test('renders buttontext properly', () => {
		const wrapper = mount(<CustomButton />);
		expect(wrapper.find('img').length).toBe(1);
	});
});
