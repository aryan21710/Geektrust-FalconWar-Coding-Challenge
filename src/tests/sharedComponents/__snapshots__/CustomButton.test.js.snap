// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING CustomButton COMPONENT Snapshot 1`] = `
<Fragment>
  <styled.div
    disabled={true}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    opacity={1}
    width="someText"
  >
    <Styled(Component)
      id="CustomButton-MiniJetImg"
      leftpos="someText"
      src="minijet.png"
      style={
        Object {
          "transform": AnimatedInterpolation {
            "attach": [Function],
            "calc": [Function],
            "children": Array [],
            "detach": [Function],
            "payload": Array [
              AnimatedValue {
                "animatedStyles": Set {},
                "children": Array [],
                "done": false,
                "lastPosition": 0,
                "lastTime": undefined,
                "lastVelocity": undefined,
                "payload": undefined,
                "setValue": [Function],
                "startPosition": 0,
                "startTime": 1604906383452,
                "value": 0,
              },
            ],
          },
        }
      }
    />
    <Styled(Component)
      data-testid="CustomButton-Text"
      style={
        Object {
          "opacity": AnimatedValue {
            "animatedStyles": Set {},
            "children": Array [],
            "done": false,
            "lastPosition": 1,
            "lastTime": undefined,
            "lastVelocity": undefined,
            "payload": undefined,
            "setValue": [Function],
            "startPosition": 1,
            "startTime": 1604906383452,
            "value": 1,
          },
        }
      }
    >
      someText
    </Styled(Component)>
  </styled.div>
</Fragment>
`;
